pkg/server/handler.go

```diff
@@ -225,6 +226,9 @@ func WithWorkspaceProjection(apiHandler http.Handler) http.HandlerFunc {
 			newPath := path.Join("/services/workspaces", cluster.Name.String(), req.URL.Path)
 			klog.V(4).Infof("Rewriting %s -> %s", path.Join(cluster.Name.Path(), req.URL.Path), newPath)
 			req.URL.Path = newPath
+			if !opts.Virtual.Enabled && opts.Extra.ShardVirtualWorkspaceURL != "" {
+				req.URL.Host = opts.Extra.ShardVirtualWorkspaceURL
```

Comment:
I think we should read the shard resource from the DB (via a lister), doing that ensures we will read default values.

Reply:
OK.

Reply:
After thinking though it again, I'm not sure it's necessary:
There are 2 cases:
- when admission will set it to the shard base URL, that means that we're in the case where it has not been specified through the CLI, probably in the case when virtual workspaces are embedded in the KCP shard. Then I don't need a URL at all since the project should just change the URL PATH (without a redirect)
- when this value is set through the option, it will be used used anyway (not overridden by the admission plugin) => let's use it directly.